Lesson02

nllbr opened the pull request:
с пятым заданием много сложностей возникло, так и не смогла разобраться. с 6 вообще не понимаю чего от меня хотят, буду думать до четверга

nllbr committed:
Create 1 task

nllbr committed:
Delete .gitignore

nllbr committed:
Delete allhw

nllbr committed:
Delete README.md

nllbr committed:
Create 2 task

nllbr committed:
Create 3 task

nllbr committed:
Create 4 task

nllbr committed:
Create 5 task

очень долго пыталась разобраться и консультировалась с ярославом пробовала разные варианты поиска и структуры программы, но у меня не вышло, ноя пыталась

ShapovaLOVALOVA commented on 1 task:
```diff
@@ -0,0 +1,12 @@
+
+list_ex = [1, 2.34, '555', None, True, ('ф', 3), {1: 'один', 2: 'два'}]
+list_res = list_ex.copy()
+i = 0
```
можно не инциировать переменную i, если она является переменной-итератором в цикле for in.

ShapovaLOVALOVA commented on 2 task:
```diff
@@ -0,0 +1,31 @@
+    element = input('введите элемент списка, для того, чтобы закончить ввод введиите "stop": ')
+    given_list.append(element)
+    if element == 'stop':
+        given_list.pop()
```
чтобы не удалять последний введенный элемент, append выполнить если element не stop (по сути в ветке else можно было сделать append)

ShapovaLOVALOVA commented on 2 task:
```diff
@@ -0,0 +1,31 @@
+    j += 1
+    if i is len(given_list):
+        break
+print(result_list)
```
Попыталась запустить этот код, к сожалению, не дал нужного результата.
Получилось только дублирование уже имеющихся элементов. 
Во-первых, что бросается в глаза: нигде не используется индекс j (но при этом вы его инкрементируете)
Во-вторых, вот такая запись некорректна: if i is len(given_list). Оператор is - оператор сравнения идентичности объектов. Эти объекты - не одно и то же. У них одинаковые значения. А если необходимо сравнить по значению, то необходимо использовать оператор ==. 
Нужно еще подумать.

ShapovaLOVALOVA commented on 4 task:
```diff
@@ -0,0 +1,16 @@
+Если в слово длинное, выводить только первые 10 букв в слове."""
+
+user_string = input('введите строку из нескольких слов, разделенными пробелами ')
+user_list = user_string.split(' ')
```
Можно функции split явно не указывать разделитель пробел. Так как по умолчанию и так произойдет разделение по пробельному символу.

nllbr replied:
да, вспомнила это уже позже)

ShapovaLOVALOVA commented on 5 task:
```diff
@@ -0,0 +1,40 @@
+                    else:
+                        def_list.insert(i, new_el)
+                        not_found = False
+        print(def_list)
```
Получился бесконечный цикл.
БОльшие и нагруженные конструкции, поэтому сами запутались.
Лучше не перегружать код а следовать принципу "разделяй и властвуй".
Этап 1. Цикл для проверки введенного элемента пользователем. Сделайте его отдельно, не нужно дальше в нем осуществлять расчеты.
Этап 2. Как только корректно завершится Этап 1 переходите к поиску места для вставки нового рейтинга.
У вас получилось 3 цикла и каждый крутится внутри другого.
На мой взгляд, цикл while not_found лишний. Вам достаточно использовать только цикл for i in и делать break если случилась вставка.
Пока не работает, как надр. Нужно докрутить еще.

nllbr committed:
Update 2 task

понравилось очень ваше решение через срезы и с учетом знаний с 3 урока сделала так

nllbr committed:
Update 4 task

nllbr committed:
Update 5 task

nllbr committed:
Create task06

что-то он у меня не очень корректно работает, помогите, пожалуйста, разобраться

ShapovaLOVALOVA commented on 2 task:
```diff
@@ -4,28 +4,12 @@
     if element == 'stop':
         given_list.pop()
+given_list[:-1:2], given_list[1::2] = given_list[1::2], given_list[:-1:2]
 print(given_list)
```
Просмотрено +

ShapovaLOVALOVA commented on 5 task:
```diff
@@ -4,37 +4,32 @@
+              f' {def_list_copy} ')
+                break
+            i += 1
+
```
Просмотено +
